tests/sentry/integrations/github/test_client.py

```diff
@@ -575,3 +589,160 @@ def assert_proxy_request(self, request, is_proxy=True):
+            )
+            in msg.body
+        )
+
+    @mock.patch("sentry.integrations.github.client.get_jwt", return_value=ApiError)
+    @responses.activate
+    def test_fatal_integration(self, get_jwt):
+        """
+        fatal fast shut off with disable flag off, integration should be broken but not disabled
+        """
+        responses.add(
+            responses.POST,
+            status=403,
+            url="https://api.github.com/graphql",
+            json={
+                "message": "This installation has been suspended",
+                "documentation_url": "https://docs.github.com/rest/reference/apps#create-an-installation-access-token-for-an-app",
+            },
+        )
+
+        self.gh_client.integration = None
+        with pytest.raises(Exception):
+            self.gh_client.get_blame_for_file(self.repo, "foo.py", "main", 1)
+        buffer = IntegrationRequestBuffer(self.gh_client._get_redis_key())
+        assert buffer.is_integration_broken() is True
+        integration = Integration.objects.get(id=self.integration.id)
+        assert integration.status == ObjectStatus.ACTIVE
+
+    @responses.activate
+    def test_error_integration(self):
+        """
+        recieve two errors and errors are recorded, integration is not broken yet so no disable
+        """
+        responses.add(
+            responses.POST,
+            status=404,
+            url="https://api.github.com/graphql",
+            json={
+                "message": "Not found",
+            },
+        )
+        responses.add(
+            responses.POST,
+            status=404,
+            url="https://api.github.com/graphql",
+            json={
+                "message": "Not found",
+            },
+        )
+        self.gh_client.integration = None
+        with pytest.raises(Exception):
+            self.gh_client.get_blame_for_file(self.repo, "foo.py", "main", 1)
+        with pytest.raises(Exception):
+            self.gh_client.get_blame_for_file(self.repo, "foo.py", "main", 1)
+        buffer = IntegrationRequestBuffer(self.gh_client._get_redis_key())
+        assert int(buffer._get_all_from_buffer()[0]["error_count"]) == 2
+        assert buffer.is_integration_broken() is False
+
+    @responses.activate
+    @with_feature("organizations:github-disable-on-broken")
+    def test_slow_integration_is_not_broken_or_disabled(self):
+        """
+        slow test with disable flag on
+        put errors and success in buffer for 10 days, assert integration is not broken or disabled
+        """
+
+        responses.add(
+            responses.POST,
+            status=404,
+            url="https://api.github.com/graphql",
+            json={
+                "message": "Not found",
+            },
+        )
+        buffer = IntegrationRequestBuffer(self.gh_client._get_redis_key())
+        now = datetime.now() - timedelta(hours=1)
```

Comment:
Is this the same way that one test was written that turned flaky? You might want to hard code the date here.